crates/harp/src/vector/mod.rs

```diff
@@ -100,12 +101,32 @@ pub trait Vector {
         Rf_xlength(self.data()) as usize
     }
 
-    fn format_one(&self, x: Self::Type) -> String;
-
-    fn format_elt_unchecked(&self, index: isize) -> String {
+    fn format_one(
+        &self,
+        x: Self::Type,
+        options: Option<&FormattedVectorCharacterOptions>,
+    ) -> String;
+
+    fn format_elt_unchecked(
+        &self,
+        index: isize,
+        options: Option<&FormattedVectorCharacterOptions>,
```

Comment:
I think we should move the definition of `FormattedVectorCharacterOptions` into this file. 
We probably also want to change it's name to something like `FormatOptions` as it's no longer specific to character vectors.